tests/tensor/rewriting/test_linalg.py

```diff
@@ -656,3 +651,15 @@ def test_cholesky_diag_from_diag():
         atol=1e-3 if config.floatX == "float32" else 1e-8,
         rtol=1e-3 if config.floatX == "float32" else 1e-8,
     )
+
+
+def test_dont_apply_cholesky():
+    x = pt.tensor("x", shape=(7, 7))
+    y = pt.eye(7, k=-1) * x
+    # Here, y is not a diagonal matrix because of k = -1
+    z_cholesky = pt.linalg.cholesky(y)
+
+    # REWRITE TEST (should not be applied)
+    f_rewritten = function([x], z_cholesky, mode="FAST_RUN")
+    nodes = f_rewritten.maker.fgraph.apply_nodes
+    assert any(isinstance(node.op, Cholesky) for node in nodes)
```

Comment:
Add a test cases where the eye is degenerate, like `y = pt.eye(1) * x`. This shouldn't trigger the rewrite.